src/main/java/mods/railcraft/RailcraftConfig.java

```diff
@@ -0,0 +1,198 @@
+        .define("boreDestorysBlocks", false);
+
+      this.boreMinesAllBlocks = builder
+        .comment("Change to false to enable mining checks, use true setting with caution, especially on servers")
+        .define("boreMinesAllBlocks", true);
+
+      this.cartsBreakOnDrop = builder
+        .comment("Change to \"true\" to restore vanilla behavior")
+        .define("cartsBreakOnDrop", false);
+
+      this.fuelPerSteamMultiplier = builder
+        .comment("Adjust the amount of fuel used to create Steam")
+        .defineInRange("fuelPerSteamMultiplier", 1.0F, 0.2F, 6.0F);
+
+      this.steamLocomotiveEfficiency = builder
+        .comment("Adjust the multiplier used when calculating fuel use")
+        .defineInRange("steamLocomotiveEfficiency", 3.0F, 0.2F, 12F);
+      builder.pop();
+    }
+  }
+
+  public static class Common {
+    public final BooleanValue enableSeasons;
+    public final IntValue christmas;
+    public final IntValue halloween;
+    public final IntValue harvest;
+
+    Common(Builder builder) {
+      builder.comment("General configuration settings")
+        .push("common");
+
+      this.enableSeasons = builder
+        .comment("Enable season-based item & train effects?")
+        .define("enableSeasons", true);
+
+      this.christmas = builder
+        .comment("Controls whether Christmas mode is (0) enabled, (1) forced, or (2) disabled")
+        .defineInRange("christmas", 0, 0, 2);
+
+      this.halloween = builder
+        .comment("Controls whether Halloween mode is (0) enabled, (1) forced, or (2) disabled")
+        .defineInRange("halloween", 0, 0, 2);
+
+      this.harvest = builder
+        .comment("Controls whether Harvest mode is (0) enabled, (1) forced, or (2) disabled")
+        .defineInRange("harvest", 0, 0, 2);
+
+      builder.pop();
+    }
+  }
+
+  public static class Client {
+    public final BooleanValue enableGhostTrain;
+    public final BooleanValue enablePolarExpress;
+    public final IntValue locomotiveLightLevel;
+
+    Client(Builder builder) {
+      builder.comment("Client only settings, mostly things related to rendering")
+        .push("client");
+
+      this.enableGhostTrain = builder
+        .comment("Change to false to disable Ghost Train rendering")
+        .define("enableGhostTrain", true);
+
+      this.enablePolarExpress = builder
+        .comment("Change to false to disable Polar Express (snow) rendering")
+        .define("enablePolarExpress", true);
+
+      this.locomotiveLightLevel = builder
+        .comment("Change '14' to a number ranging from '0' to '15' to represent the dynamic lighting of the locomotive when Dynamic Lights mod is present.\nIf it is '0' then locomotive lightning will be disabled.")
+        .defineInRange("locomotiveLightLevel", 14, 0, 15);
+
+      builder.pop();
+    }
+  }
+
+  public static final ForgeConfigSpec clientSpec;
+  public static final Client CLIENT;
```

Comment:
Use lower case as instances of `Client` aren't constants. Use this as a guidance for when to use all upper case:
1. All `static final` primitive types (Remember that all interface fields are inherently static final).
2. All `static final` object reference types that are never followed by "`.`" (dot).
3. All `static final` arrays that are never followed by "`[`" (opening square bracket).